src/aliased_buffer.h

```diff
@@ -231,29 +220,26 @@ class AliasedBuffer {
   void reserve(size_t new_capacity) {
     DCHECK_GE(new_capacity, count_);
     DCHECK_EQ(byte_offset_, 0);
-    DCHECK(free_buffer_);
     const v8::HandleScope handle_scope(isolate_);
 
     const size_t old_size_in_bytes = sizeof(NativeT) * count_;
     const size_t new_size_in_bytes = sizeof(NativeT) * new_capacity;
 
+    // allocate v8 new ArrayBuffer
+    v8::Local<v8::ArrayBuffer> ab = v8::ArrayBuffer::New(
+        isolate_, new_size_in_bytes);
+
     // allocate new native buffer
-    NativeT* new_buffer = Calloc<NativeT>(new_capacity);
+    NativeT* new_buffer = static_cast<NativeT*>(ab->GetContents().Data());
```

Comment:
auto here ?

Reply:
We've made a decision about being explicit if the type is not too verbose: https://github.com/nodejs/node/blob/master/CPP_STYLE_GUIDE.md#using-auto

Reply:
But in current codebase, it's already used this way
https://github.com/nodejs/node/blob/c70e85372343ab445676cf2115f65a4e44f97c94/src/node_platform.cc#L235

Reply:
Also used by v8 by default
https://github.com/nodejs/node/blob/9b4bf7de6c9a7c25f116c7a502384c20b5cfaea3/deps/v8/.clang-tidy#L7

Reply:
@gengjiawen The example in `node_platform.cc` you’re referring to uses a more complex type – I would agree with using `auto` for that. Generally, we’re not being super consistent here, and we’ve had debates about this before. ;)

I think `auto` would be okay here too, because the `static_cast` makes things a bit more obvious, but I still think `NativeT*` is simple enough as a type to be kept this way.

(Basically, I agree with what the linked clang docs are saying, I just think `NativeT*` is short enough for me.)

Reply:
There is an option `MinTypeNameLength`, disscuss it's value to make `auto` rule more consistent ?
https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-auto.html#options

Reply:
@gengjiawen It’s just my opinion, but my gut feeling would be something like 10 or 12 characters or so, and only to use `auto` when the type is obvious (e.g. from `new Something()` or when using a cast).